### PR TITLE
Fix: [regression] com_joomlaupdate throws Notice: Undefined property: JUpdate::$downloadurl

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #9401 and #9392.
#### Summary of Changes

This forward port the changes we made in 2.5.x to 3.x if we have no download URL: https://github.com/joomla/joomla-cms/commit/73f560137fba04cf41394dbb725050bebf964080
relevant are only that changes to `administrator/components/com_joomlaupdate/views/default/tmpl/default.php` and `administrator/language/en-GB/en-GB.com_joomlaupdate.ini` the others are backeportes from 3.x to 2.5.x that day.
#### Testing Instructions
1. Install last staging / Nightly build: https://developer.joomla.org/cms-packages/
2. switch the core update URL to "Test"
3. go to com_joomlaupdate
#### Expected Result

You should get a message that there are no download URL availiable.

### CODE REVIEW

**Comment by zero-24** [comment]
@brianteeman can you have a look into the language string. I guess it needs improving or are you ok with it?

File: administrator/language/en-GB/en-GB.com_joomlaupdate.ini [hunk 1]
```diff
@@ -27,6 +27,8 @@ COM_JOOMLAUPDATE_UPDATE_LOG_URL="Downloading update file from %s."
 COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
 COM_JOOMLAUPDATE_VIEW_COMPLETE_MESSAGE="Your site has been successfully updated. Your Joomla version is now %s."
 COM_JOOMLAUPDATE_VIEW_DEFAULT_DOWNLOAD_IN_PROGRESS="Downloading update file. Please wait ..."
+COM_JOOMLAUPDATE_VIEW_DEFAULT_NO_DOWNLOAD_URL="We can't find a download URL"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_joomlaupdate.ini]: There are two possibilities why this may happen.

Comment (zero-24) [inline administrator/language/en-GB/en-GB.com_joomlaupdate.ini]: Thanks fixed with: https://github.com/joomla/joomla-cms/commit/14f69f9b9b6cfb1c5a9b5496ab76d513ebd8762c

**Comment by infograf768** [comment]
available. not availiable I guess.

**Comment by brianteeman** [comment]
@infograf768 is correct - hard to see on my phone :)

**Comment by zero-24** [comment]
fixed with https://github.com/joomla/joomla-cms/commit/1b257a3e80e4c23d85791bc916bb48753ecb5510 Thanks.

**Comment by mbabker** [comment]
I still say these patches are only bandages over the real issue and that's inside the `JUpdater` library.  That comment on the other issue about the update still showing on the control panel supports that theory.  `JUpdater` should NOT be presenting an update if it does not have a fully valid record, and what's causing these undefined index errors is the fact that the update definition is not complete/valid for 3.5.  To be presented an update when using a collection update stream, you should need both a valid entry in the base collection XML file (our list.xml) AND the extension XML file (our extension.xml).  Right now, Joomla treats only the entry in the collection XML file as enough to say an update is available.

**Comment by zero-24** [comment]
hmm there is one condition where this can happen. And this is the `We don't support your PHP Version`. There you have a valid entry for the list.xml and for the extension.xml but as the update don't support your PHP Version there is no download URL available. But we want to show the message `Hey please update your PHP to get more updates.` ;)

**Comment by mbabker** [comment]
The extension XML file would still have a valid update definition for that version though.  Right now, there isn't a valid update definition for 3.5 across the board.  https://update.joomla.org/core/test/list_test.xml has a potential update for 3.5 users listed, but https://update.joomla.org/core/teststs/extension_sts.xml does not and that's why JUpdater is getting confused; since the latter doesn't even have a supporting update for 3.5, it should negate the former file saying "hey, there's an update available".

Also, the PHP minimum version check is already handled in the [extension update adapter](https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/updater/adapters/extension.php#L108) so it shouldn't need to be duplicated into the update component.

File: administrator/language/en-GB/en-GB.com_joomlaupdate.ini (hunk 1 above)
Comment (andrepereiradasilva) [inline administrator/language/en-GB/en-GB.com_joomlaupdate.ini]: "An update was found. But is not possible to update your Joomla installation"?

File: administrator/language/en-GB/en-GB.com_joomlaupdate.ini [hunk 2]
```diff
@@ -27,6 +27,8 @@ COM_JOOMLAUPDATE_UPDATE_LOG_URL="Downloading update file from %s."
...
+COM_JOOMLAUPDATE_VIEW_DEFAULT_NO_DOWNLOAD_URL_DESC="We can't find the dwnload URL to %s. There are two possibilities why this may happen. The first is your host doesn't support <a href="_QQ_"http://www.joomla.org/technical-requirements.html"_QQ_">the minimum requirements for Joomla %s</a> and there is no alternative available. Or there is a problem with the Update XML files on the Joomla Update Server."
```
Comment (andrepereiradasilva) [inline administrator/language/en-GB/en-GB.com_joomlaupdate.ini]: "An update to Joomla %s was found, but it wasn't possible to fetch the download URL for that update.<br />There are two possibilities why this happens:<br />- Your host doesn't support <a href="_QQ_"http://www.joomla.org/technical-requirements.html"_QQ_">the minimum requirements for Joomla %s</a>.<br />- There is a problem with Joomla Update Server."

**Comment by andrepereiradasilva** [comment]
This works, just made some comments to the text.

File: administrator/language/en-GB/en-GB.com_joomlaupdate.ini (hunk 2 above)
Comment (richard67) [inline administrator/language/en-GB/en-GB.com_joomlaupdate.ini]: Typo "dwnload" in this text.

**Pull request closed without merging**